src/jsrm/systems/planar_pcs_num.py

```diff
@@ -0,0 +1,1305 @@
+        """
+        Add a small number to the bending strain to avoid singularities
+        """
+        xi_reshaped = xi.reshape((-1, 3))
+
+        xi_bend_sign = jnp.sign(xi_reshaped[:, 0])
+        
+        # set zero sign to 1 (i.e. positive)
+        xi_bend_sign = jnp.where(xi_bend_sign == 0, 1, xi_bend_sign)
+        
+        # add eps to the bending strain (i.e. the first column)
+        sigma_b_epsed = lax.select(
+            jnp.abs(xi_reshaped[:, 0]) < _eps,
+            xi_bend_sign * _eps,
+            xi_reshaped[:, 0],
+        )
+        xi_epsed = jnp.stack(
+            [
+                sigma_b_epsed,
+                xi_reshaped[:, 1],
+                xi_reshaped[:, 2],
+            ],
+            axis=1,
+        )
+
+        # Flatten the array
+        xi_epsed = xi_epsed.flatten()
+
+        return xi_epsed
+    
+    def classify_segment(
+        params: Dict[str, Array], 
+        s: Array
+        ) -> Tuple[Array, Array]:
+        """
+        Classify the point along the robot to the corresponding segment.
+
+        Args:
+            params (Dict[str, Array]): dictionary of robot parameters
+            s (Array): point coordinate along the robot in the interval [0, L].
+
+        Returns:
+            Tuple[Array, Array]: 
+                - segment_idx (Array): index of the segment where the point is located
+                - s_segment (Array): point coordinate along the segment in the interval [0, l_segment]
+        """
+        l = params["l"]
+        
+        # Compute the cumulative length of the segments starting with 0
+        l_cum = jnp.cumsum(jnp.concatenate([jnp.array([0.0]), l]))
+        
+        # Classify the point along the robot to the corresponding segment
+        segment_idx = jnp.clip(jnp.sum(s > l_cum) - 1, 0, len(l) - 1)
+        
+        # Compute the point coordinate along the segment in the interval [0, l_segment]
+        s_segment = s - l_cum[segment_idx]
+
+        return segment_idx, s_segment.squeeze(), l_cum
+    
+    @jit
+    def chi_fn_xi(
+        params: Dict[str, Array], 
+        xi: Array, 
+        s: Array
+    ) -> Array:
+        """
+        Compute the pose of the robot at a given point along its length with respect to the strain vector.
+
+        Args:
+            params (Dict[str, Array]): dictionary of robot parameters.
+            xi (Array): strain vector of the robot.
+            s (Array): point coordinate along the robot in the interval [0, L].
+
+        Returns:
+            Array: pose of the robot at the point s in the interval [0, L].
+        """
+        th0 = jnp.array(params["th0"])  # initial angle of the robot
+        l = params["l"] # length of each segment [m]
+        
+        # Classify the point along the robot to the corresponding segment
+        segment_idx, s_local, _ = classify_segment(params, s)
+            
+        chi_O = jnp.array([0.0, 0.0, th0])  # Initial pose of the robot
```

Comment:
I think you should use `jnp.concatenate([jnp.zeros((2, )), th0])` instead of constructing a new `jnp.array`. This could possibly have a bad influence on compilation and evaluation time as you are calling the class constructor again on a variable that is already a jax array.